src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,164 @@
+      switch (sortBy) {
+        case 'name':
+        case 'sex':
+          return sortOrder === 'asc'
+            ? firstPerson[sortBy].localeCompare(secondPerson[sortBy])
+            : secondPerson[sortBy].localeCompare(firstPerson[sortBy]);
+
+        case 'born':
+        case 'died':
+          return sortOrder === 'asc'
+            ? +firstPerson[sortBy] - +secondPerson[sortBy]
+            : +secondPerson[sortBy] - +firstPerson[sortBy];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  const applyQuery = useCallback(
+    debounce((newQuery: string) => {
+      if (newQuery) {
+        searchParams.set('query', newQuery.toLowerCase().trimLeft());
+      } else {
+        searchParams.delete('query');
+      }
+
+      history(`?${searchParams.toString()}`);
+    }, 500),
+    [],
+  );
+
+  const queryHandler: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const { value } = e.target;
+
+    setQuery(value);
+    applyQuery(value);
+  };
+
+  const sortHandler: MouseEventHandler<HTMLTableHeaderCellElement> = (e) => {
+    const value = e.currentTarget.textContent?.toLowerCase().replace('*', '');
+
+    if (value) {
+      searchParams.set('sortBy', value);
+    }
+
+    if (sortOrder === 'asc') {
+      searchParams.set('sortOrder', 'desc');
+    } else {
+      searchParams.set('sortOrder', 'asc');
+    }
+
+    history(`?${searchParams.toString()}`);
+  };
+
+  return (
+    <>
+      <input
+        type="search"
+        name="searsh"
+        value={query || ''}
+        placeholder="Enter name"
+        onChange={queryHandler}
+      />
+
+      <table className="PeopleTable">
+        <thead>
+          <tr className="PeopleTable__Header">
+            <th
+              className="PeopleTable__Title PeopleTable__Title--sort"
+              onClick={sortHandler}
+            >
+              {sortBy === 'name' ? 'Name*' : 'Name'}
+            </th>
+
+            <th
+              className="PeopleTable__Title PeopleTable__Title--sort"
+              onClick={sortHandler}
+            >
+              {sortBy === 'sex' ? 'Sex*' : 'Sex'}
+            </th>
+
+            <th
+              className="PeopleTable__Title PeopleTable__Title--sort"
+              onClick={sortHandler}
+            >
+              {sortBy === 'born' ? 'Born*' : 'Born'}
+            </th>
+
+            <th
+              className="PeopleTable__Title PeopleTable__Title--sort"
+              onClick={sortHandler}
+            >
+              {sortBy === 'died' ? 'Died*' : 'Died'}
+            </th>
+
+            <th className="PeopleTable__Title">Mother</th>
+
+            <th className="PeopleTable__Title">Father</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {visiblePeople.map(person => {
+            const personUrl = `${person.name.toLowerCase().replace(/ /g, '-')}-${person.born}`;
+
+            return (
+              <tr
+                key={person.name}
```

Comment:
not sure whether it is a good idea to set a person.name as a key value, as there might be 2 or even more people with the same names in the list